ecommerce/frontend/src/components/checkout/EditCardModal.js

```diff
@@ -0,0 +1,157 @@
+            },
+            sm: {
+                span: 18,
+            },
+        },
+    }
+
+    const handleInputFocus = val => setFocused(val.target.name)
+
+    return (
+        <Modal
+            visible={visible}
+            title="Edit your credit card information"
+            destroyOnClose
+            onOk={() => {
+                setVisible(false)
+            }}
+            onCancel={() => {
+                setVisible(false)
+            }}
+            cancelText="Go Back"
+            okText="Edit">
+            <Form
+                {...formItemLayout}
+                layout="horizontal"
+                form={form}
+                name="editCardForm"
+                onFinish={() => {}}
+                onFinishFailed={() => {}}
+                scrollToFirstError
+                initialValues={{ cardName: cardName, ...cardProps }}>
+                <Form.Item
+                    name="cardName"
+                    label="Card Name"
+                    rules={[
+                        {
+                            required: true,
+                            message: 'Please input your card name!',
+                        },
+                    ]}>
+                    <Input
+                        onChange={val => {
+                            setCardNickname(val.target.value)
+                        }}
+                    />
+                </Form.Item>
+                <Form.Item
+                    name="name"
+                    label="Name"
+                    rules={[
+                        {
+                            required: true,
+                            message: 'Please input your card name!',
+                        },
+                    ]}>
+                    <Input
+                        onFocus={handleInputFocus}
+                        onChange={val => {
+                            setName(val.target.value)
+                        }}
+                        name="name"
+                    />
+                </Form.Item>
+                <Form.Item
+                    name="number"
+                    label="Card Number"
+                    rules={[
+                        {
+                            required: true,
+                            message: 'Please input your card number!',
+                        },
+                    ]}>
+                    <Input
+                        onFocus={handleInputFocus}
+                        onChange={val => {
+                            setNumber(val.target.value)
+                        }}
+                        name="number"
+                    />
+                </Form.Item>
+                <Form.Item
+                    name="expiry"
+                    label="Expiry Date"
+                    rules={[
+                        {
+                            required: true,
+                            message: 'Please input your card expiry date!',
+                        },
```

Comment:
Add stricter validation on all fields. Also assist the user by adding `/` in SKT field for example and `-` in card number